nautobot/docs/development/apps/api/platform-features/populating-extensibility-features.md

```diff
@@ -59,3 +63,61 @@ config = AnimalSoundsConfig
+
+## Advanced
+
+While the signal handler approach works for most use cases, it does have its downsides. Other options for the population of base state are available, each with their own trade-offs. There is no single best option, you have to choose the one has the best properties for your specific need.
+
+- Signal Handlers on `nautobot_database_ready` (see above)
+- [Django Onboard Tooling](https://docs.djangoproject.com/en/4.2/howto/initial-data/)
+    - [Django Data Migrations](https://docs.djangoproject.com/en/4.2/topics/migrations/#data-migrations)
+    - [Django Fixtures](https://docs.djangoproject.com/en/4.2/topics/db/fixtures/#fixtures)
+- [Design Builder Jobs](https://docs.nautobot.com/projects/design-builder/en/latest/)
+- Creating data in-place where needed
+
+The following table details points that are somewhat easily comparable between the different options.
+
+| Option                                       | Startup Impact                       | Execution | Performance | Re-running | Idempotence | Modifications Possible      | Custom App Needed |
+|----------------------------------------------|--------------------------------------|-----------|-------------|------------|-------------|-----------------------------|-------------------|
+| Signal Handlers on `nautobot_database_ready` | Exceptions prevent container startup | Automatic | Average     | Possible   | Enforced    | Overwrite/crash on next run | Yes               |
+| Django Data Migrations                       | Exceptions prevent container startup | Automatic | Average     | Impossible | N/A         | Yes                         | Yes               |
+| Django Fixtures                              | N/A                                  | Manual    | Good        | Possible   | TBD         | Overwrite/Crash on next run | No                |
+| Design Builder Jobs                          | N/A                                  | Manual    | Average     | Possible   | Possible    | Overwrite/Crash on next run | No                |
+| Creating Data In-place Where Needed          | N/A                                  | Automatic | Average     | Possible   | Possible    | Overwrite/crash on next run | No                |
+
+The following sections detail additional points that weren't possible to fit.
+
+### Signal Handlers On `nautobot_database_ready`
+
+- ➕ Can be updated in-place in later App releases
+- Re-runs every time `nautobot-server migrate` or `nautobot-server post_upgrade` is run
+    - ➖ Must be idempotent, as such it mustn't introduce duplicate records or errors if run repeatedly
+    - ➖ May recreate or revert records that a user intentionally deleted or altered - if this is not desired it has to be accounted for and handled in the code
+- ➖ Impacts Nautobot startup and upgrade performance, care needs to be taken
```

Comment:
```suggestion
- ➖ Impacts Nautobot startup and upgrade performance, care needs to be taken
- ➖ May require updates over time to remain operational as Nautobot's data model evolves
```

(Alternately, this probably should be another item in the table above as it's a factor for many of these approaches)

Reply:
I went with the alternative approach.